src/pages/en/guides/markdown-content.md

```diff
@@ -103,14 +106,67 @@ An example blog post `content` object might look like:
 ```
 
 :::note
-`astro`, `file`, and `url` are the only guaranteed properties provided by Astro in the `content` prop. The rest of the object is defined by your frontmatter variables.
+`astro`, `file`, and `url` are the only guaranteed properties provided by Astro in the Markdown `content` prop. The rest of the object is defined by your frontmatter variables.
+
+📚 Read more about the [MDX content prop](/en/guides/integrations-guide/mdx/).
+
 :::
 
+#### Layouts for multiple page types
+
+An Astro layout can receive both the content object from `.md` and `.mdx` files, as well as any named props passed from `.astro` files.
+
+In the example below, the layout will display the page title either from an Astro component passing a `title` attribute, or from a frontmatter YAML `title` property:
+
+```astro
+---
+//src/components/MyLayout.astro
+const { content, title } = Astro.props
+
+const pageTitle = title || content.title
+---
+<html>
+  <head></head>
+  <body>
+    <h1>{pageTitle}</h1>
+    <slot />
+  </body>
+</html>
+```
+
+ALTERNATE EXAMPLE - CHOOSE WHICH ONE WE WANT, IF EITHER?
+
+```astro
+---
+// src/components/MyLayout.astro
+const { title } = Astro.props || Astro.props.content
+---
+<html>
+  <head></head>
+  <body>
+    <h1>{title}</h1>
+    <slot />
+  </body>
+</html>
+```
```

Comment:
My preference would be for the second of you two examples just because it’s a slightly cleaner and easier to extend format (you can easily add `description` or whatever additional frontmatter stuff just by adding to the destructuring).

One note: we should switch `props` and `props.content` otherwise `props.content` will never get used:

````suggestion
```astro
---
// src/components/MyLayout.astro
const { title } = Astro.props.content || Astro.props;
---
<html>
  <head></head>
  <body>
    <h1>{title}</h1>
    <slot />
  </body>
</html>
```
````

There is some nuance here but I think if the aim is to grab stuff from frontmatter, this pattern works well 👍